test/ParameterStoreBuildWrapperPluginTest.groovy

```diff
@@ -84,5 +125,38 @@ class ParameterStoreBuildWrapperPluginTest {
             assertEquals(ParameterStoreBuildWrapperPlugin.class, result)
         }
     }
+
+    class withGlobalParameter {
+        @After
+        public void reset() {
+            ParameterStoreBuildWrapperPlugin.reset()
+        }
+
+        @Test
+        void addGlobalParameter() {
+            def result = ParameterStoreBuildWrapperPlugin.withGlobalParameter('/path/', [])
+
+            assertEquals([[path: '/path/']], result.globalParameterOptions)
+        }
```

Comment:
I like this `addGlobalParameter` method.  You're exercising that when `withGlobalParameter` is called, the given paramter is added to the list.  

I might make one modification.  `withGlobalParameter('/path/', [])` <-- is this how you *want* other people to call `withGlobalParameter`, when they have no options?  You've define options to be optional `withGlobalParameter(String path, options=[])`, which means you could call it *without* the empty array.  

Part of the nice effect of unit tests is illustrating how you expect and want others to use your code.  This is an opportunity to illustrate the  convenience of the method you designed.  You  could  call your method  with  `withGlobalParameter('/path/')`

Reply:
resolved to show both no options passed and empty options passed as well